benchmark/unified_benchmark.py

```
@@ -0,0 +1,1170 @@
+        """Initialize all benchmarking components"""
+        try:
+            # Initialize template manager
+            template_config = TemplateConfig(
+                template_dir="benchmark/test_templates",
+                cache_dir=os.path.join(self.config.output_dir, "template_cache")
+            )
+            self.template_manager = TemplateManager(template_config)
+            self.logger.info("Template manager initialized")
+            
+            # Initialize test generator
+            test_gen_config = TestGenerationConfig(
```

Ruff (0.12.2)

Undefined name `TestGenerationConfig`

(F821)